src/main/java/bio/terra/cli/businessobject/resource/GcsObject.java

```diff
@@ -125,6 +131,26 @@ public String resolve(boolean includeUrlPrefix) {
         : bucketName + "/" + objectName;
   }
 
+  /**
+   * Check if the GCS bucket object is a directory by checking for the presence of a child object
+   * with a trailing slash. There will always be an object present to represent the directory/prefix
+   * itself.
+   */
+  public boolean isDirectory() throws SystemException {
+    try {
+      BucketCow bucketCow =
+          CrlUtils.createStorageCow(Context.requireUser().getPetSACredentials()).get(bucketName);
+      Iterable<BlobCow> objects =
+          bucketCow
+              .list(BlobListOption.currentDirectory(), BlobListOption.prefix(objectName))
+              .getValues();
+      return StreamSupport.stream(objects.spliterator(), false)
+          .anyMatch(object -> object.getBlobInfo().isDirectory());
```

Comment:
Oh thanks! It would behave the same that way.

So I went with listing the objects with the prefix because I needed to do something similar in the UI to handle a rare edge case where a directory created manually in gcs does not get recognized as a directory. Just tested that case and looks like I wasn't handling it anyways. Making a change to handle nonempty user created directories. It's not possible to handle empty folders created by users since its ambigious given what we have (false for isDirectory, 1 object in the list which is the same for any regular file objects).